public/pages/workflows/workflow_list/resource_list.tsx

```diff
@@ -0,0 +1,207 @@
+      setItemIdToExpandedRowMap(updatedItemIdToExpandedRowMap);
+    } else {
+      try {
+        const result = await fetchResourceData(item, dataSourceId!, dispatch);
+        if (result) {
+          setCodeBlockData({ item, data: result });
+        }
+      } catch (error) {
+        console.error('Failed to fetch resource data:', error);
+      }
+    }
+  };
+
+  // Handles table pagination and sorting.
+  const onTableChange = ({
+    page = { index: pageIndex, size: pageSize },
+    sort = { field: 'id', direction: 'asc' },
+  }: any) => {
+    setPageIndex(page.index);
+    setPageSize(page.size);
+    setSortField(sort.field as keyof WorkflowResource);
+    setSortDirection(sort.direction);
+  };
+
+  const sortedItems = [...allResources].sort((a, b) => {
+    const multiplier = sortDirection === 'asc' ? 1 : -1;
+    return a[sortField] > b[sortField]
+      ? 1 * multiplier
+      : a[sortField] < b[sortField]
+      ? -1 * multiplier
+      : 0;
+  });
+
+  const pagination = {
+    pageIndex,
+    pageSize,
+    totalItemCount: allResources.length,
+    pageSizeOptions: [10, 25, 50],
+  };
+
+  return (
+    <EuiFlexGroup direction="column">
+      <EuiFlexItem>
+        <EuiBasicTable
+          itemId="id"
+          itemIdToExpandedRowMap={itemIdToExpandedRowMap}
+          isExpandable
+          sorting={{ sort: { field: sortField, direction: sortDirection } }}
+          pagination={pagination}
+          onChange={onTableChange}
+          columns={[
+            ...columns,
+            {
+              align: RIGHT_ALIGNMENT,
+              width: '40px',
+              isExpander: true,
+              render: (item: WorkflowResource) => (
+                <EuiButtonIcon
+                  onClick={() => toggleDetails(item)}
+                  aria-label={
+                    itemIdToExpandedRowMap[item.id] ? 'Collapse' : 'Expand'
+                  }
+                  iconType={
+                    itemIdToExpandedRowMap[item.id] ? 'arrowUp' : 'arrowDown'
+                  }
+                />
+              ),
+            },
+          ]}
+          items={sortedItems}
+          noItemsMessage={'No existing resources found'}
+        />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+}
+
+// Fetches Resource details data for a given resource item.
+export const fetchResourceData = async (
+  item: WorkflowResource,
+  dataSourceId: string,
+  dispatch: any
+) => {
+  if (item.stepType.includes('ingest_pipeline')) {
+    return await dispatch(
+      getIngestPipeline({ pipelineId: item.id, dataSourceId })
+    ).unwrap();
+  } else if (item.stepType.includes('index')) {
+    return await dispatch(
+      getIndex({
+        index: item.id,
+        dataSourceId,
+      })
+    ).unwrap();
+  } else if (item.stepType.includes('search_pipeline')) {
+    return await dispatch(
+      getSearchPipeline({
+        pipelineId: item.id,
+        dataSourceId,
+      })
+    ).unwrap();
+  }
```

Comment:
Please use the `WORKFLOW_STEP_TYPE` constants in common/interfaces.ts to determine which dispatch calls to execute.